ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -250,6 +275,43 @@ object WebSocketService {
           import spray.json._, JsonProtocol._
           "matchedQueries" -> p.toJson
         }
+
+      override def splitRequest(
+          prefix: Option[domain.StartingOffset],
+          request: SearchForeverRequest,
+      ): (Option[SearchForeverRequest], Option[SearchForeverRequest]) = {
+
+        // If a prefix is defined, it's substituted to any empty offset in the request
+        val adjustedRequest =
+          prefix.fold(request)(prefix =>
+            request.copy(
+              queries = request.queries.map(query =>
+                query.copy(offset = query.offset.orElse(Some(prefix.offset)))
+              )
+            )
+          )
+
+        val (withoutOffset, withOffset) =
+          adjustedRequest.queries.toList.partition(_.offset.isEmpty)
+
+        val toAcs = list.toNel(withoutOffset).map(SearchForeverRequest(_))
+        val toTxs = list.toNel(withOffset).map(SearchForeverRequest(_))
+
+        (toAcs, toTxs)
+      }
+
+      import scalaz.syntax.tag._
+      private implicit val stringOrder = scalaz.Order.fromScalaOrdering[String]
+
+      override def startingOffset(
+          prefix: Option[domain.StartingOffset],
+          request: SearchForeverRequest,
+      ): Option[domain.StartingOffset] =
+        request.queries
+          .map(_.offset)
+          .minimumBy(_.fold("")(_.unwrap))
+          .get // Safe on NonEmptyList
```

Comment:
Pull in `foldable1` syntax and use `minimumBy1` instead.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/9847/commits/7670eadebd735de629897ad66bdada2fb876d455